### PR TITLE
refactor: clean up css pertaining to overview

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

Cleans up css pertaining to overview. Required some refactoring around styles in details view generally (which is good since I got to remove some general styles there).

I **did** validate other h2 and h3 elements that would be impacted by the details view changes and things seemed good. This is also why I added the font-size change to the collapsible h2 element.

Before:
image

After **(note: help text)**:
image

Before:
image

After **(note: instances header)**:
image

Before:
image

After:
image

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by peterdur** [comment]
`yarn test:e2e overview` fails on my Windows box and in the e2e_web_mac job

File: src/DetailsView/components/overview-content/overview-content-container.tsx [hunk 1]
```diff
@@ -78,9 +78,9 @@ export const OverviewContainer = NamedFC<OverviewContainerProps>('OverviewContai
     );
 
     return (
-        <div className="overview">
+        <div className={styles.overview}>
```
Comment (peterdur) [inline src/DetailsView/components/overview-content/overview-content-container.tsx:81]: Test failure seems to be a result of test not being updated to reflect this change - overview.test.ts@37 is still looking for `overviewSelectors.overview` which is `'.overview'`, but now the style has a generated id like `'overview--1tASe'`

Comment (waabid) [reply src/DetailsView/components/overview-content/overview-content-container.tsx]: Yep! Just needed to change to automation id.

Comment (peterdur) [reply src/DetailsView/components/overview-content/overview-content-container.tsx]: Just verified on my box that's it, got the test to pass using automation id.

**Pull request merged**